### PR TITLE
Adding extra trace logging for User-Agent

### PR DESCRIPTION
**Pull request by bdemers**

### CODE REVIEW

**Comment by velo** [comment]
+1

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/StatelessAndStatefulWebSessionManager.java [hunk 1]
```diff
@@ -57,15 +57,26 @@ public class StatelessAndStatefulWebSessionManager
...
+        if( WebUtils.isHttp( context ) )
         {
-            create( session );
+            HttpServletRequest request = WebUtils.getHttpRequest( context );
+
+            if ( isStatelessClient( request ) )
+            {
+                // we still need to set the session id, WHY?
+                ( (SimpleSession) session ).setId( fakeSessionIdGenerator.generateId( session ) );
+                log.debug( "Stateless client sesion {} is not persisted.", session.getId() );
```
Comment (nabcos) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/StatelessAndStatefulWebSessionManager.java]: typo

**Comment by nabcos** [comment]
+1

File: nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/StatelessAndStatefulWebSessionManager.java (hunk 1 above)
Comment (bdemers) [inline nexus/nexus-rest-api/src/main/java/org/sonatype/nexus/security/StatelessAndStatefulWebSessionManager.java]: good eye!
It's fixed

**Comment by peterlynch** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
What for is this one needed?

**Comment by bdemers** [comment]
Improved logging to help trouble shoot problems.

**Pull request merged**